hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieKeyLookupHandle.java

```diff
@@ -46,13 +48,13 @@ public class HoodieKeyLookupHandle<T, I, K, O> extends HoodieReadHandle<T, I, K,
   private static final Logger LOG = LoggerFactory.getLogger(HoodieKeyLookupHandle.class);
 
   private final BloomFilter bloomFilter;
-  private final List<String> candidateRecordKeys;
+  private final Set<String> candidateRecordKeys;
   private long totalKeysChecked;
 
   public HoodieKeyLookupHandle(HoodieWriteConfig config, HoodieTable<T, I, K, O> hoodieTable,
                                Pair<String, String> partitionPathFileIDPair) {
     super(config, hoodieTable, partitionPathFileIDPair);
-    this.candidateRecordKeys = new ArrayList<>();
+    this.candidateRecordKeys = new HashSet<>();
```

Comment:
This set is passed in to `filterKeysFromFile` which used to take this list and immediately convert to a set so I have updated this to be a set from the start.